### PR TITLE
Add husky and lint-staged

### PR DESCRIPTION
**Pull request by DecampsRenan**
- Add pre-commit hook to run prettier on staged files
- Fix `pnpm build` command (ignore `./old` folder to make the astro build and check pass)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,7 +7,11 @@
     "start": "astro dev",
     "build": "astro check && astro build",
     "preview": "astro preview",
-    "astro": "astro"
+    "astro": "astro",
+    "prepare": "husky"
+  },
+  "lint-staged": {
+    "*.{ts,tsx,js,jsx,json}": "prettier --write"
```
[the remark this answers is not in the document]
Comment (DecampsRenan) [reply package.json:14]: good catch, i don't know if prettier know how to parse these files (i suppose it doesn't without the proper plugin, maybe we need to check astro plugins to see if there is something we can use)

[the remark this answers is not in the document]
Comment (DecampsRenan) [reply package.json:14]: i found this plugin: https://github.com/withastro/prettier-plugin-astro

i will try to setup it tomorrow if i can :)

[the remark this answers is not in the document]
Comment (yoannfleurydev) [reply package.json:14]: The plugin is already in place, just used by IDEs, not the scripting part. I'll check that.

**Review by yoannfleurydev — APPROVED** [review]

**Comment by yoannfleurydev** [comment]
Thanks ❤️

**Pull request merged**